scripts/fetch/utils.R

```diff
@@ -0,0 +1,93 @@
+#' @title get Map data as sf
+#' 
+#' @description Gets data from \code{maps::map()} and returns a transformed sf object
+#' 
+#' @param crs sf package crs string. If included, data will be transformed to the given crs
+#' @param within sf polygon. polygons completely outside the polygon will not be returned.
+#' Can be retrieved from the \code{bbox_to_polygon()} function.
+#' @param ... inputs to maps::map. For example: \code{database = "state"} or 
+#' \code{database = "world", region = "Puerto Rico"}.
+#' 
+#' @return \code{data.frame} of class \code{sf} with geometry and ID columns.
+#' 
+#' @example 
+#' states <- get_map_data_sf(crs="+init=epsg:5070", database="state")
+#' puerto_rico <- get_map_data_sf(crs="+init=epsg:5070", database = "world", region = "Puerto Rico")
+#' counties <- get_map_data_sf(crs="+init=epsg:3857", database = "county")
+#' maps inputs xlim and ylim can limit the response to be within a lat/lon bounding rectangle
+#' islands <- get_map_data_sf(crs = "+init=epsg:3857", database = 'mapdata::world2Hires', 
+#'                            regions = "(?!USA)", xlim = c(275, 300), ylim = c(16, 30))
+#'
+#'# A polygon in a particular projection can also be used.
+#' bbox <- bbox_to_polygon(c(-87, 21, -70, 34), return_crs = "+init=epsg:3857")
+#' storm_counties <- get_map_data_sf(database = "counties", crs = "+init=epsg:3857", within = bbox)
+get_map_data_sf <- function(..., crs=NULL, within = NULL){
```

Comment:
It might be nice to have the default for `crs` be the crs of `within`:

Otherwise, how do you do the `within` if you don't specify a crs and the crs of `within` isn't WGS84?

Reply:
You would get an error in that case. I prefer not to have default projections because people should be declaring them rather than making assumptions. (akin to knowing what units your data are in -- defaults are dangerous).